sdk-core/src/main/java/care/data4life/sdk/wrapper/SdkFhirParser.kt

```diff
@@ -18,9 +18,9 @@ internal object SdkFhirParser : WrapperContract.FhirParser {
     }
 
     override fun toFhir4(resourceType: String, source: String): Fhir4Resource {
-        val clazz = fhirElement.getFhir4ClassForType(resourceType)!!
+        val clazz = fhirElement.getFhir4ClassForType(resourceType)
 
-        return fhir4Parser.toFhir(clazz, source)
+        return fhir4Parser.toFhir(clazz!!, source)
```

Comment:
if `fhirElement.getFhir4ClassForType(resourceType)` could produce null, it makes sense to handle this case explicit. Using `!!` creates a NullPointerException but the Stacktrace for `!!` is not helpfull to debug later, no line number.

Reply:
`!!` should only be used if Java Kotlin interop incorrectly reports a nullable type but it never could be null. If the value could be null, becuase of the programm logic and there needs to be a null handling instead.

Reply:
This is just alignment with its twin method. `null` will only happen, if a consumer seeks to parse a DomainResource or other Resource we do not currently support.
However in principle I agree that we should tackle this in better way, but I see it in a separate task since this PR does not change the current behaviour in this particular case. Do you agree?

Reply:
I commented, because the issue was just moved from on property to the other